src/jit/importer.cpp

```diff
@@ -4118,12 +4123,21 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
         }
     }
 
-#if FEATURE_HW_INTRINSICS && defined(_TARGET_XARCH_)
+#if FEATURE_HW_INTRINSICS
+#if defined(_TARGET_XARCH_)
     if ((namespaceName != nullptr) && strcmp(namespaceName, "System.Runtime.Intrinsics.X86") == 0)
     {
         InstructionSet isa = lookupHWIntrinsicISA(className);
         result             = lookupHWIntrinsic(methodName, isa);
     }
+#elif defined(_TARGET_ARM64_)
+    if ((namespaceName != nullptr) && strcmp(namespaceName, "System.Runtime.Intrinsics.Arm.Arm64") == 0)
+    {
+        result = lookupHWIntrinsic(className, methodName);
+    }
```

Comment:
We could agree on a common interface here.  Move namespace name inside.

`lookupHWIntrinsic(namespaceName, className, methodName)`

This allows IsSupported and Platform not supported exceptions to be handled in JIT.

Reply:
This could be converted into switch this way as well

Reply:
Ex. start with `System.Runtime.Intrinsics`, switch on `X86`, `Arm`, and than next switch statements if possible,

Reply:
I was thinking we should eventually be using an unordered/hash dictionary lookup

Reply:
Actually there is one preference - to get it as fast as possible. @tannergooding and @fiigii were thinking about dictionary or switch, whichever would better work here. From this what I see it could be split into couple of switches or as you suggest lookup could be dictionary based.